walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,117 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+  private ArrayList<TimeRange> getMergedTimes(ArrayList<TimeRange> takenTimes) {
+    Collections.sort(takenTimes, TimeRange.ORDER_BY_START);
+    ArrayList<TimeRange> takenTimesMerged = new ArrayList();
+    if (takenTimes.size() != 0) {
+      TimeRange previous = takenTimes.get(0);
+
+      for (TimeRange time : takenTimes) {
+        if (time.overlaps(previous)) {
+          int start = previous.start();
+          int end = Math.max(time.end(), previous.end());
+          previous = TimeRange.fromStartEnd(start, end, false);
+        } else {
+          takenTimesMerged.add(previous);
+          previous = time;
+        }
+      }
+      takenTimesMerged.add(previous);
+    }
+    return takenTimesMerged;
+  }
+
+  private ArrayList<TimeRange> getBetweenTimes(ArrayList<TimeRange> takenTimesMerged) {
+    int END_OF_DAY = 1440;
```

Comment:
Laura found TimeRange.START_OF_DAY and TimeRange.END_OF_DAY, so looks like you don't need to define your own variables : )

Reply:
TimeRange.END_OF_DAY will not work for my case due to the fact that they have it set to 23 hours and 59 minutes rather than just 24. Thus my options are to keep what I have or do (TimeRange.END_OF_DAY + 1) @LukeGMTravis

Reply:
Which do you think is better?

Reply:
Ah I see. I think personally I'd still prefer relying on TimeRange.END_OF_DAY but it's not that big a deal